sql/core/src/main/scala/org/apache/spark/sql/execution/joins/ShuffledHashJoinExec.scala

```diff
@@ -57,6 +57,8 @@ case class ShuffledHashJoinExec(
 
   override def outputOrdering: Seq[SortOrder] = joinType match {
     case FullOuter => Nil
+    case LeftOuter if buildSide == BuildLeft => Nil
+    case RightOuter if buildSide == BuildRight => Nil
```

Comment:
can we add some comments to explain why the ordering can't be preserved?

Reply:
Sure, added a comment as per my understanding (please let me know if I misunderstand something)

My thought was, because the second iteration on the build-side (for outer-join semantic) is on a hashedRelation, the result cannot be in order.